ir/src/main/scala/org/scalajs/ir/Types.scala

```diff
@@ -163,17 +163,41 @@ object Types {
     }
   }
 
+  sealed abstract class NonArrayTypeRef extends TypeRef
+
+  /** Primitive type reference. */
+  final case class PrimRef private[ir] (tpe: Type)(
+      private[ir] val encodedName: String, private[ir] val tag: Int)
+      extends NonArrayTypeRef {
+
+    override def equals(that: Any): Boolean = this eq that.asInstanceOf[AnyRef]
+
+    override def hashCode(): Int = tag
+  }
+
+  final val VoidRef = PrimRef(NoType)("V", Tags.TagVoidRef)
+  final val BooleanRef = PrimRef(BooleanType)("Z", Tags.TagBooleanRef)
+  final val CharRef = PrimRef(CharType)("C", Tags.TagCharRef)
+  final val ByteRef = PrimRef(ByteType)("B", Tags.TagByteRef)
+  final val ShortRef = PrimRef(ShortType)("S", Tags.TagShortRef)
+  final val IntRef = PrimRef(IntType)("I", Tags.TagIntRef)
+  final val LongRef = PrimRef(LongType)("J", Tags.TagLongRef)
+  final val FloatRef = PrimRef(FloatType)("F", Tags.TagFloatRef)
+  final val DoubleRef = PrimRef(DoubleType)("D", Tags.TagDoubleRef)
+  final val NullRef = PrimRef(NullType)("N", Tags.TagNullRef)
+  final val NothingRef = PrimRef(NothingType)("E", Tags.TagNothingRef)
```

Comment:
I think it would make sense to provide an accessor per `encodedName` here, just like in an enum. The split of this logic seems a bit strange.

The same applies to tags.